File: problems/typescript/1.two-sum.ts
/*
 * @lc app=leetcode id=1 lang=typescript
 *
 * [1] Two Sum
 */

// @lc code=start
function twoSum(nums: number[], target: number): number[] | void {
  const map: Map<number, number> = new Map();

  for (let i = 0; i < nums.length; i++) {
    const match = map.get(target - nums[i]);
    if (match != undefined) return [match, i];
    map.set(nums[i], i);
  }
}
// @lc code=end

File: problems/kotlin/1.two-sum.kt
/*
 * @lc app=leetcode id=1 lang=kotlin
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf<Int, Int>()

        for ((index, num) in nums.withIndex()) {
            val match = map.get(target - num)
            if (match != null) {
                return intArrayOf(match, index)
            }
            map.set(num, index)
        }

        return intArrayOf()
    }
}
// @lc code=end

